### ISSUE TITLE
Value.Create does not work with Type.Object({date: Type.Date({ default: new Date() })})

### ISSUE THREAD

**Issue by itsyoboieltr** [sinclairzx81/typebox#670]
Hey, love your work! I have run into the following problem:

I am trying to generate objects from a typebox schema. According to my intuition, the default value for a generated date would be:

```ts
const object = Value.Create(Type.Date())
console.log(object);
// expected: new Date()
```

According to the current implementation (from: [create.ts](https://github.com/sinclairzx81/typebox/blob/master/src/value/create.ts#L132))
```ts
const object = Value.Create(Type.Date())
console.log(object);
// output: new Date(0)
```

I would guess this is for reasons of consistency, so that the output of the Value.Create() function is always the same. 

In my application, I would always want to generate an object with the current date. I saw that there is the `default` option, where I could provide my own value.

```ts
const object = Value.Create(Type.Date({ default: new Date() }))
console.log(object);
// output: new Date()
```

And this worked wonderfully! I was so happy I could make it work according to my needs. Then I proceeded to put this as a key of an object.

```ts
const object = Value.Create(Type.Object({date: Type.Date()}));
console.log(object);
// output: { date: new Date(0) }
```

This again returned not the current date, as I did not specify the default date. I added the default date.

```ts
const object = Value.Create(Type.Object({date: Type.Date({ default: new Date() })}));
console.log(object);
// output: { date: {} }
```

This somehow returns an empty object, instead of the Date itself. I wonder what went wrong here?

**Comment by sinclairzx81** [comment]
@itsyoboieltr Hi, Thanks for reporting.

I have pushed out a fix for this on 0.31.27. There are two changes in this revision

- Fix for nested object default value (`{ date: {} }`)
- Create dates with the current time by default

On the second change, I've actually updated the date returned on Value.Create to be `new Date()` (giving the current time, not the unix epoch timestamp 0). You're right that TypeBox initialized 0 for the timestamp as this was a predictable initial value, but I do think there is value in initializing the current time, mostly because this is what you get back from a Date constructor and that there is no other mechanism (not even using `default`) to initialize with the current time.

```typescript
const date = Value.Create(Type.Date()) // current time

const date = new Date()                // current time
```
This can be overriden with the following
```typescript
const date = Value.Create(Type.Date({default: new Date(0) )) // unix epoch
```
Let me know if that resolves things
Cheers
S
